extractors/log/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+use error::RuntimeError;
+use shared::async_nats::{self};
+use shared::clap;
+use shared::clap::Parser;
+use shared::log;
+use shared::nats_subjects::Subject;
+use shared::prost::Message;
+use shared::protobuf::event_msg::EventMsg;
+use shared::protobuf::event_msg::event_msg::Event;
+use shared::protobuf::log_extractor;
+use shared::tokio::fs::File;
+use shared::tokio::io::{AsyncBufReadExt, BufReader};
+use shared::tokio::sync::watch;
+
+mod error;
+
+/// TODO: Update
+/// The peer-observer log-extractor periodically queries data from the
+/// Bitcoin Core Log endpoint and publishes the results as events into
+/// a NATS pub-sub queue.
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Args {
+    /// Address of the NATS server where the extractor will publish messages to.
+    #[arg(short, long, default_value = "127.0.0.1:4222")]
+    pub nats_address: String,
+
+    /// The log level the extractor should run with. Valid log levels are "trace",
+    /// "debug", "info", "warn", "error". See https://docs.rs/log/latest/log/enum.Level.html.
+    #[arg(short, long, default_value_t = log::Level::Debug)]
+    pub log_level: log::Level,
+}
+
+impl Args {
+    pub fn new(nats_address: String, log_level: log::Level) -> Args {
+        Self {
+            nats_address,
+            log_level,
+        }
+    }
+}
+
+pub async fn run(args: Args, mut shutdown_rx: watch::Receiver<bool>) -> Result<(), RuntimeError> {
+    log::debug!("Connecting to NATS server at {}..", args.nats_address);
+    let nats_client = async_nats::connect(&args.nats_address).await?;
+    log::info!("Connected to NATS server at {}", &args.nats_address);
+
+    log::info!("Opening bitcoind log pipe...");
+    let file = File::open("bitcoind_pipe").await?;
```

Comment:
You'll probably get to this at a later point: we'd probably want the location of the pipe to be configurable via an `Args` entry.